samples/contrib/pytorch-samples/test_compile_yamls.py

```diff
@@ -0,0 +1,424 @@
+                    # For GPU set gpu count and accelerator type
+                    gpus=0,
+                    accelerator="None",
+                ).after(prep_task).set_display_name("Training")
+            )
+
+            minio_tb_upload = (
+                COMPONENT_MNIO(
+                    bucket_name="mlpipeline",
+                    folder_name=log_dir,
+                    input_path=train_task.outputs["tensorboard_root"],
+                    filename="",
+                ).after(train_task
+                       ).set_display_name("Tensorboard Events Pusher")
+            )
+            minio_mar_upload = (
+                COMPONENT_MNIO(
+                    bucket_name="mlpipeline",
+                    folder_name=mar_path,
+                    input_path=train_task.outputs["checkpoint_dir"],
+                    filename="bert_test.mar",
+                ).after(train_task).set_display_name("Mar Pusher")
+            )
+            minio_config_upload = (
+                COMPONENT_MNIO(
+                    bucket_name="mlpipeline",
+                    folder_name=config_prop_path,
+                    input_path=train_task.outputs["checkpoint_dir"],
+                    filename="config.properties",
+                ).after(train_task).set_display_name("Conifg Pusher")
+            )
+
+            model_uri = str(model_uri)
+            isvc_yaml = """
+            apiVersion: "serving.kubeflow.org/v1beta1"
+            kind: "InferenceService"
+            metadata:
+              name: {}
+              namespace: {}
+            spec:
+              predictor:
+                serviceAccountName: sa
+                pytorch:
+                  storageUri: {}
+                  resources:
+                    limits:
+                      memory: 4Gi   
+            """.format(deploy, namespace, model_uri)
+
+            # For GPU inference use below yaml with gpu count and accelerator
+            gpu_count = "1"
+            accelerator = "nvidia-tesla-p4"
+            isvc_gpu_yaml = """
+            apiVersion: "serving.kubeflow.org/v1beta1"
+            kind: "InferenceService"
+            metadata:
+              name: {}
+              namespace: {}
+            spec:
+              predictor:
+                serviceAccountName: sa
+                pytorch:
+                  storageUri: {}
+                  resources:
+                    limits:
+                      memory: 4Gi   
+                      nvidia.com/gpu: {}
+                  nodeSelector:
+                    cloud.google.com/gke-accelerator: {}
+        """.format(deploy, namespace, model_uri, gpu_count, accelerator)
+            # Update inferenceservice_yaml for GPU inference
+            deploy_task = (
+                COMPONENT_DEPLOY(
+                    action="apply", inferenceservice_yaml=isvc_yaml
+                ).after(minio_mar_upload).set_display_name("Deployer")
+            )
+
+        compiler.Compiler().compile(
```

Comment:
How are the failure cases being caught?

Reply:
@chauhang The google cloud pipeline compilation test which I referred to didn't include any negative cases. I'll try to add negative cases for this.